test/integration/scheduler/util.go

```diff
@@ -166,7 +198,7 @@ func initTestSchedulerWithOptions(
 	t *testing.T,
 	context *testContext,
 	setPodInformer bool,
-	policy *schedulerapi.Policy,
+	policy string,
```

Comment:
Is there any reason that it passes `policy string` instead of `policy *schedulerapi.Policy` as an argument? If using `policy *schedulerapi.Policy`, it could avoid a lot of code change.

Reply:
yes `schedulerapi.Policy` can't be used to create `schedulerconfig.SchedulerAlgorithmSource` to be passed to the scheduler constructor. So, it's either changing it into `schedulerapiv1 "k8s.io/kubernetes/pkg/scheduler/api/v1"` in order to be decoded into a string and then create the algorithm provider, or changing it into a string.

Reply:
How about converting `*schedulerapi.Policy` to `string` in the func `createSchedulerWithPodInformer` or `createAlgorithmProviderFromPolicyString`?

Reply:
I couldn't find any codec that supports encoding `*schedulerapi.Policy` (that's why I mentioned changing it to `schedulerapiv1`). Also marshaling it into JSON would be some what messy (specially when it comes to nested structs) as we need to change first letter of each field to lowercase.

Reply:
This will lead to lots of changes, where we would need to change it in the `factory` structs and all the related structs.